src/interface/iimpl.rs

```diff
@@ -393,76 +418,134 @@ impl IfaceImpl {
             }
         }
         for field in &self.metadata {
+            dup_metadata
+                .entry(field.name.clone())
+                .and_modify(|counter| *counter += 1)
+                .or_insert(0);
             if !schema.meta_types.contains_key(&field.id) {
                 errors.push(ImplInconsistency::SchemaMetaAbsent(field.name.clone(), field.id));
             }
         }
 
+        dup_metadata.iter().for_each(|(field_name, &count)| {
+            if count > 1 {
+                errors.push(ImplInconsistency::RepeatedMetaData(field_name.clone(), count));
+            }
+        });
```

Comment:
This (and below) can be expressed more "rusty":
```suggestion
        dup_metadata.iter().filter(|(_, &count)| count > 1).for_each(|(field_name, &count)| {
            errors.push(ImplInconsistency::RepeatedMetaData(field_name.clone(), count));
        });
```